DrillProcess.py

```diff
@@ -23,215 +19,143 @@
 ROTATION_SPEED = 2e3                # at 12 volts
 SAMPLE_HOLDER_HEIGHT = 100
 STOP_ABOVE_GROUND = 1 # mm
+DEADZONE = 0.2 #analog joysticks arent perfect
+TRIG_DEADZONE = 0.05
 
-### INITIALIZTION ###
+done = False
+
+
+'''### INITIALIZTION ###
 drill.storage.top_motor_movement = 0
 drill.storage.bottom_motor_movement = 0
 drill.storage.rotating = False
 drill.storage.top_distance = 0
 drill.storage.bottom_distance = 0
-drill.storage.carousel_position = 'EMPTY'
-
-### FUNCTIONS ###
-async def top_is_moving():
-    if drill.storage.top_motor_movement == 1:
-        return 'RAISING'
-    elif drill.storage.top_motor_movement == 2:
-        return 'LOWERING'
+drill.storage.carousel_position = 'EMPTY'''
+
+### CONSTANTS ###
+DRILL_RAISE_DUTY_CYCLE = 0.2*1e5    # at 12 volts
+DRILL_LOWER_DUTY_CYCLE = 0.1*1e5    # at 12 volts
+ROTATION_SPEED = 2e3                # at 12 volts
+SLEEP_DURATION = 0.1 #seconds
+controller_num = config.science_controller
+
+### INITIALIZTION ###
+ScienceDevice = Device('ScienceModule', 'rover', network=config.network)
+motor_movement_types = ['stopped', 'raising', 'lowering']
+
+ScienceDevice.storage.top_motor_movement = top_move
+ScienceDevice.storage.bottom_motor_movement = bot_move
+ScienceDevice.storage.rotation_direction = rot_move
+ScienceDevice.storage.carousel_position = 'home'
+ScienceDevice.storage.ready = 0
+ScienceDevice.storage.top_raise_switch = 0
+ScienceDevice.storage.top_lower_switch = 0
+ScienceDevice.storage.bottom_raise_switch = 0
+ScienceDevice.storage.bottom_lower_switch = 0
+ScienceDevice.storage.sample_switch = 0
+ScienceDevice.storage.empty_switch = 0
+
+top_move = 'stopped'
+bot_move = 'stopped'
+rot_move = 'stopped'
+
+
+@ScienceDevice.on('*/controller{}/joystick1'.format(controller_num))
+async def top_motor_movement(joystick1, data):
+    global top_move
+    axis = data[1]
+
+    if axis is None:
+        return
+    else:
+        duty_cycle = axis*DRILL_RAISE_DUTY_CYCLE
+
+    if -DEADZONE < axis < DEADZONE:
+        top_move = 'stopped'
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(0)})
+    elif axis > DEADZONE:
+        top_move = 'raising'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(duty_cycle)})
+    elif axis < -DEADZONE:
+        top_move = 'lowering'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(duty_cycle)})
     else:
-        return 'STOPPED'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(0)})
 
-async def bottom_is_moving():
-    if drill.storage.bottom_motor_movement == 1:
-        return 'RAISING'
-    elif drill.storage.bottom_motor_movement == 2:
-        return 'LOWERING'
+    #ScienceDevice.top_motor_movement = top_move
+    log.debug('setting top movement to {}'.format(top_move))
+    await ScienceDevice.sleep(SLEEP_DURATION)
```

Comment:
We don't need to sleep the device in these callbacks, that's already handled by the JoystickProcess.